nucleus/modelci/__init__.py

```diff
@@ -0,0 +1,295 @@
+        """Retrieves info of the Unit Test.
+
+        Args:
+            unit_test_id: ID of Unit Test
+
+        Returns:
+            A UnitTestInfo object
+        """
+        response = self._connection.make_request(
+            {},
+            f"modelci/unit_test/{unit_test_id}/info",
+            requests_command=requests.get,
+        )
+        return UnitTestInfo(**response)
+
+    def list_unit_tests(self) -> List[UnitTest]:
+        """Lists all Unit Tests of the current user. ::
+
+            import nucleus
+            from nucleus.modelci.unit_test import ThresholdComparison
+            client = nucleus.NucleusClient("YOUR_SCALE_API_KEY")
+            unit_test = client.modelci.create_unit_test(
+                "sample_unit_test", "slc_bx86ea222a6g057x4380"
+            )
+
+            client.modelci.list_unit_tests()
+
+        Returns:
+            A list of UnitTest objects.
+        """
+        response = self._connection.make_request(
+            {},
+            "modelci/unit_test",
+            requests_command=requests.get,
+        )
+        return [
+            UnitTest(test_id, self) for test_id in response["unit_test_ids"]
+        ]
+
+    def get_unit_test_metrics(self, unit_test_id: str) -> List[UnitTestMetric]:
+        """Retrieves all metrics of the Unit Test. ::
+
+            import nucleus.modelci as nm
+            client = nucleus.NucleusClient("YOUR_SCALE_API_KEY")
+            unit_test = client.modelci.list_unit_tests()[0]
+
+            client.modelci.get_unit_test_metrics(unit_test.id)
+
+        Args:
+            unit_test_id: ID of Unit Test
+
+        Returns:
+            A list of UnitTestMetric objects
+        """
+        response = self._connection.make_request(
+            {},
+            f"modelci/unit_test/{unit_test_id}/metrics",
+            requests_command=requests.get,
+        )
+        return [
+            UnitTestMetric(**metric)
+            for metric in response["unit_test_metrics"]
+        ]
+
+    def delete_unit_test(self, unit_test_id: str) -> bool:
+        """Creates a new Unit Test. ::
+
+            import nucleus
+            from nucleus.modelci.unit_test import ThresholdComparison
+            client = nucleus.NucleusClient("YOUR_SCALE_API_KEY")
+            unit_test = client.modelci.list_unit_tests()[0]
+
+            success = client.modelci.create_unit_test(unit_test.id)
+
+        Args:
+            unit_test_id: unique ID of unit test
+
+        Returns:
+            Whether deletion was successful.
+        """
+        response = self._connection.make_request(
+            {},
+            f"modelci/unit_test/{unit_test_id}",
+            requests_command=requests.delete,
+        )
+        return response["success"]
+
+    def evaluate_model_on_unit_tests(
+        self, model_id: str, unit_test_names: List[str]
+    ) -> AsyncJob:
+        """Evaluates the given model on the specified Unit Tests. ::
```

Comment:
Documentation is slightly incorrect here, if the unit_test_names are omitted, the model will be evaluated on all unit tests.  The array of names is a subsampling method, if that makes sense

Reply:
Hmm I think that no matter what, the model will only be evaluated on the specified unit tests, since `unit_test_names` is not optional.

In the case that it's the empty list, we would pass an empty list into the `TestEvaluator` constructor, and it looks to me like the subsampling is only checking if the input list is null, not if it's empty (see https://github.com/scaleapi/scaleapi/blob/d994d223ca552aee814f7c8fadbe26a6e6ee49bb/server/models/nucleus/modelci/MCUnitTest.ts#L209-L211)

Reply:
Opened up `node` just to confirm
```
> tests = []
[]
> if (tests) console.log("hello")
hello
```